components/script/dom/xrsession.rs

```diff
@@ -395,7 +398,10 @@ impl XRSession {
         // Step 3: XXXManishearth handle inline session
 
         // Step 4-5
-        let mut callbacks = mem::replace(&mut *self.raf_callback_list.borrow_mut(), vec![]);
+        mem::replace(
```

Comment:
I think here we should instead just swap current and raf_callback_list, and after the for loop we should clear current

Reply:
This basically ensures we don't hold on to RAF callbacks longer than necessary

Reply:
OK! Is that `drop` necessary?

Reply:
Yes, because otherwise the guard will be borrowed for too long. You can also do this by inserting an enclosing scope.